packages/shared/src/server/auth/invalidateApiKeys.ts

```diff
@@ -0,0 +1,110 @@
+import { prisma } from "../../db";
+import { redis, safeMultiDel } from "..";
+import { logger } from "../logger";
+
+import { type ApiKey } from "../../db";
+
+/**
```

Comment:
I feel like bundling the rejection with auth is odd. IMO we could have a separate lookup + cache for this behaviour which would remove the need to invalidate API keys from Redis from the worker.

Having a separate condition check could make the behaviour more obvious, especially since it's only used for ingestion. E.g. prompt management should never need the relevant new property.